File: articles/values-strivings-and-the-direction-of-personality/r/analyze_values_strivings_direction.R
root <- normalizePath(file.path(dirname(sys.frame(1)$ofile), ".."))
data_path <- file.path(root, "data", "synthetic_values_strivings_direction.csv")
outputs <- file.path(root, "outputs")
dir.create(outputs, showWarnings = FALSE, recursive = TRUE)

data <- read.csv(data_path, stringsAsFactors = FALSE)

data$transcendence_level <- ifelse(
  data$self_transcendence > median(data$self_transcendence),
  "higher_transcendence",
  "lower_transcendence"
)
data$enhancement_level <- ifelse(
  data$self_enhancement > median(data$self_enhancement),
  "higher_enhancement",
  "lower_enhancement"
)
data$conflict_level <- ifelse(
  data$striving_conflict > median(data$striving_conflict),
  "higher_conflict",
  "lower_conflict"
)
data$direction_profile <- paste(
  data$transcendence_level,
  data$enhancement_level,
  data$conflict_level,
  sep = "_"
)

data$high_conflict_low_ownership <- data$striving_conflict > median(data$striving_conflict) &
  data$striving_ownership < median(data$striving_ownership)
data$high_status_low_meaning <- data$striving_status > median(data$striving_status) &
  data$striving_meaning < median(data$striving_meaning)

context_summary <- aggregate(
  data[, c("self_transcendence", "self_enhancement", "openness_to_change", "conservation", "value_tension_total", "motivational_quality", "striving_conflict", "life_direction_coherence", "life_satisfaction")],
  by = list(value_context = data$value_context),
  FUN = mean
)

profile_summary <- aggregate(
  data[, c("self_transcendence", "self_enhancement", "motivational_quality", "striving_conflict", "life_direction_coherence", "life_satisfaction")],
  by = list(direction_profile = data$direction_profile),
  FUN = mean
)

corr_cols <- c(
  "benevolence", "universalism", "self_direction", "achievement", "power", "security", "tradition", "stimulation",
  "self_transcendence", "self_enhancement", "openness_to_change", "conservation",
  "value_tension_total", "striving_meaning", "striving_status", "striving_care",
  "striving_autonomy", "striving_competence", "striving_relatedness", "striving_conflict",
  "striving_ownership", "motivational_quality", "life_direction_coherence", "life_satisfaction"
)

model_life_satisfaction <- lm(
  life_satisfaction ~ self_transcendence + self_enhancement +
    openness_to_change + conservation + striving_meaning +
    striving_status + striving_care + motivational_quality +
    striving_conflict,
  data = data
)

model_buffer <- lm(
  life_satisfaction ~ striving_conflict * motivational_quality +
    self_transcendence + self_enhancement +
    openness_to_change + conservation,
  data = data
)

model_direction <- lm(
  life_direction_coherence ~ self_transcendence +
    openness_to_change + striving_ownership +
    striving_conflict + value_tension_total,
  data = data
)

write.csv(context_summary, file.path(outputs, "r_values_strivings_context_summary.csv"), row.names = FALSE)
write.csv(profile_summary, file.path(outputs, "r_values_strivings_direction_profile_summary.csv"), row.names = FALSE)
write.csv(cor(data[, corr_cols], use = "pairwise.complete.obs"), file.path(outputs, "r_values_strivings_correlations.csv"))
capture.output(summary(model_life_satisfaction), file = file.path(outputs, "r_life_satisfaction_model_summary.txt"))
capture.output(summary(model_buffer), file = file.path(outputs, "r_conflict_buffer_model_summary.txt"))
capture.output(summary(model_direction), file = file.path(outputs, "r_life_direction_coherence_model_summary.txt"))
write.csv(data, file.path(outputs, "r_scored_values_strivings_direction.csv"), row.names = FALSE)
cat("Wrote R outputs to:", outputs, "\n")

File: articles/values-strivings-and-the-direction-of-personality/python/analyze_values_strivings_direction.py
#!/usr/bin/env python3
from pathlib import Path
import json
import numpy as np
import pandas as pd

ROOT = Path(__file__).resolve().parents[1]
DATA = ROOT / "data" / "synthetic_values_strivings_direction.csv"
OUT = ROOT / "outputs"
OUT.mkdir(exist_ok=True)

VALUES = [
    "benevolence",
    "universalism",
    "self_direction",
    "achievement",
    "power",
    "security",
    "tradition",
    "stimulation",
]
STRIVINGS = [
    "striving_meaning",
    "striving_status",
    "striving_care",
    "striving_autonomy",
    "striving_competence",
    "striving_relatedness",
    "striving_conflict",
    "striving_ownership",
]
DERIVED = [
    "self_transcendence",
    "self_enhancement",
    "openness_to_change",
    "conservation",
    "value_tension_self_transcendence_enhancement",
    "value_tension_openness_conservation",
    "value_tension_total",
    "striving_prosocial_orientation",
    "motivational_quality",
    "life_direction_coherence",
]

def ols(df, y, xs):
    X = np.column_stack([np.ones(len(df)), df[xs].to_numpy(float)])
    beta, *_ = np.linalg.lstsq(X, df[y].to_numpy(float), rcond=None)
    return dict(zip(["intercept", *xs], map(float, beta)))

df = pd.read_csv(DATA)
required = {"person_id", "value_context", "life_satisfaction", *VALUES, *STRIVINGS, *DERIVED}
missing = required - set(df.columns)
if missing:
    raise ValueError(f"Missing columns: {sorted(missing)}")

df["transcendence_level"] = np.where(df["self_transcendence"] > df["self_transcendence"].median(), "higher_transcendence", "lower_transcendence")
df["enhancement_level"] = np.where(df["self_enhancement"] > df["self_enhancement"].median(), "higher_enhancement", "lower_enhancement")
df["conflict_level"] = np.where(df["striving_conflict"] > df["striving_conflict"].median(), "higher_conflict", "lower_conflict")
df["direction_profile"] = df["transcendence_level"] + "_" + df["enhancement_level"] + "_" + df["conflict_level"]

df["high_conflict_low_ownership"] = (
    (df["striving_conflict"] > df["striving_conflict"].median())
    & (df["striving_ownership"] < df["striving_ownership"].median())
)
df["high_status_low_meaning"] = (
    (df["striving_status"] > df["striving_status"].median())
    & (df["striving_meaning"] < df["striving_meaning"].median())
)

context_summary = (
    df.groupby("value_context")
    .agg(
        n=("person_id", "count"),
        self_transcendence_mean=("self_transcendence", "mean"),
        self_enhancement_mean=("self_enhancement", "mean"),
        openness_to_change_mean=("openness_to_change", "mean"),
        conservation_mean=("conservation", "mean"),
        value_tension_mean=("value_tension_total", "mean"),
        motivational_quality_mean=("motivational_quality", "mean"),
        striving_conflict_mean=("striving_conflict", "mean"),
        life_direction_coherence_mean=("life_direction_coherence", "mean"),
        life_satisfaction_mean=("life_satisfaction", "mean"),
    )
    .reset_index()
)

profile_summary = (
    df.groupby("direction_profile")
    .agg(
        n=("person_id", "count"),
        self_transcendence_mean=("self_transcendence", "mean"),
        self_enhancement_mean=("self_enhancement", "mean"),
        motivational_quality_mean=("motivational_quality", "mean"),
        striving_conflict_mean=("striving_conflict", "mean"),
        life_direction_coherence_mean=("life_direction_coherence", "mean"),
        life_satisfaction_mean=("life_satisfaction", "mean"),
    )
    .reset_index()
)

pattern_summary = pd.DataFrame({
    "pattern": ["high_conflict_low_ownership", "high_status_low_meaning"],
    "n": [int(df["high_conflict_low_ownership"].sum()), int(df["high_status_low_meaning"].sum())],
    "proportion": [float(df["high_conflict_low_ownership"].mean()), float(df["high_status_low_meaning"].mean())],
})

models = {
    "life_satisfaction": ols(
        df,
        "life_satisfaction",
        [
            "self_transcendence",
            "self_enhancement",
            "openness_to_change",
            "conservation",
            "striving_meaning",
            "striving_status",
            "striving_care",
            "motivational_quality",
            "striving_conflict",
        ],
    ),
    "life_direction_coherence": ols(
        df,
        "life_direction_coherence",
        [
            "self_transcendence",
            "openness_to_change",
            "striving_ownership",
            "striving_conflict",
            "value_tension_total",
        ],
    ),
    "motivational_quality": ols(
        df,
        "motivational_quality",
        [
            "striving_autonomy",
            "striving_competence",
            "striving_relatedness",
            "striving_ownership",
            "striving_conflict",
        ],
    ),
    "note": "Synthetic demonstration only; not moral ranking, clinical assessment, personality testing, workplace screening, legal evaluation, or individual prediction.",
}

corr_cols = VALUES + STRIVINGS + DERIVED + ["life_satisfaction"]

context_summary.to_csv(OUT / "python_values_strivings_context_summary.csv", index=False)
profile_summary.to_csv(OUT / "python_values_strivings_direction_profile_summary.csv", index=False)
pattern_summary.to_csv(OUT / "python_values_strivings_pattern_summary.csv", index=False)
df[corr_cols].corr().to_csv(OUT / "python_values_strivings_correlations.csv")
(OUT / "python_values_strivings_models.json").write_text(json.dumps(models, indent=2))
df.to_csv(OUT / "python_scored_values_strivings_direction.csv", index=False)
print(f"Wrote outputs to {OUT}")
